json_ref_dict/loader.py

```diff
@@ -2,29 +2,65 @@
 import os
 import pathlib
 import posixpath
+from collections import deque
 from functools import lru_cache
 import json
 import mimetypes
 from os import getcwd, path
-from typing import Any, Callable, Dict, IO
+from typing import Any, Callable, Dict, IO, Deque
 from urllib.parse import urlparse
 from urllib.request import urlopen
 
 from json_ref_dict.exceptions import DocumentParseError
 
 
-CONTENT_LOADER: Callable[[IO], Dict]
+JSONSchema = Dict[str, Any]
+Parser = Callable[[IO], JSONSchema]
+DocumentLoader = Callable[[str], JSONSchema]
 
 try:
     import yaml
 
-    CONTENT_LOADER = yaml.safe_load
+    CONTENT_PARSER: Parser = yaml.safe_load
 except ImportError:  # pragma: no cover
-    CONTENT_LOADER = json.load  # pragma: no cover
+    CONTENT_PARSER: Parser = json.load  # type: ignore
+
+
+class Loader:
```

Comment:
Can I suggest we make the two attributes (`loaders` and `default`) of this class private? I would want to be able to e.g. change the loaders from a deque to something else without a breaking change, if needed in future

Reply:
Done